internal/gateway/gateway.go

```diff
@@ -0,0 +1,225 @@
+package gateway
```

Comment:
There's no new code here. It's just the merge of what we had in `mesa.go` and `system/sqlstore.go`